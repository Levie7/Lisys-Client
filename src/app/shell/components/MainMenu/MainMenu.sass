@import 'src/style/lisys/settings/settings.color'

.MainMenu
    &_DarkDrawer
        .drawer-handle
            background: $primary

            &-icon
                background: $white

                &:after,
                &:before
                    background: $white

    &_LightDrawer
        .drawer-handle
            background: $white

            &-icon
                background: $primary

                &:after,
                &:before
                    background: $primary
