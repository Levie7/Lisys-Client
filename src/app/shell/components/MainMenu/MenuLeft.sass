.MenuLeft
    &_Logo
        height: 64px