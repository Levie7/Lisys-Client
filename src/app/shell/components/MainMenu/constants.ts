export const logo = {
    dark: {
        alt: 'Lisys-Dark',
        title: 'Lisys Dark',
        url: require('./assets/logo-dark.png'),
    },
    light: {
        alt: 'Lisys-Light',
        title: 'Lisys Light',
        url: require('./assets/logo-light.png'),
    },
};

export const menuData = [
    {
        icon: 'setting',
        key: 'Settings',
        name: 'Settings',
        status: 'Active',
        url: '/settings',
    },
    {
        icon: 'dashboard',
        key: 'Dashboard',
        name: 'Dashboard',
        status: 'Active',
        url: '/',
    },
    {
        children: [
            {
                icon: 'medicine_box',
                key: 'Medicine',
                name: 'Medicine',
                status: 'Active',
                url: '/medicine',
            },
            {
                icon: 'uom',
                key: 'Unit of Measurement',
                name: 'Unit of Measurement',
                status: 'Active',
                url: '/uom',
            },
            {
                icon: 'category',
                key: 'Category',
                name: 'Category',
                status: 'Active',
                url: '/category',
            },
            {
                icon: 'variant',
                key: 'Variant',
                name: 'Variant',
                status: 'Active',
                url: '/variant',
            },
            {
                icon: 'supplier',
                key: 'Supplier',
                name: 'Supplier',
                status: 'Active',
                url: '/supplier',
            },
        ],
        icon: 'hdd',
        key: 'Master',
        name: 'Master',
        status: 'Active',
        url: '/master',
    },
    {
        children: [
            {
                icon: 'list',
                key: 'Purchase List',
                name: 'Purchase List',
                status: 'Active',
                url: '/purchase_list',
            },
            {
                icon: 'history',
                key: 'Buy Price History',
                name: 'Buy Price History',
                status: 'Active',
                url: '/buy_price_history',
            },
            {
                icon: 'payment',
                key: 'Purchase Payment',
                name: 'Purchase Payment',
                status: 'Active',
                url: '/purchase_payment',
            },
            {
                icon: 'return',
                key: 'Purchase Return',
                name: 'Purchase Return',
                status: 'Active',
                url: '/purchase_return',
            },
        ],
        icon: 'transaction',
        key: 'Purchasing',
        name: 'Purchasing',
        status: 'Active',
        url: '/purchasing',
    },
];
