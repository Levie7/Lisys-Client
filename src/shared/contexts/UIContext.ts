import * as React from 'react';

export interface UIContextValue {
    isMobile: boolean;
}

const UIContext = React.createContext<UIContextValue>({
    isMobile: true,
});
export default UIContext;
export const useUIContext = () => React.useContext(UIContext);
