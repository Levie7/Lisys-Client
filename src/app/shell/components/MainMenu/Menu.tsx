import { Menu as AntMenu } from 'antd';
import * as React from 'react';

import { Menu as MenuAPI } from 'src/core/api';

import { Icon } from 'src/shared/components/Icon';
import { Link } from 'src/shared/components/Link';

import { menuData } from './constants';

interface MenuProps {
    className?: string;
    mode?: 'vertical' | 'vertical-left' | 'vertical-right' | 'horizontal' | 'inline';
    theme?: 'light' | 'dark';
}

export const Menu = React.memo<MenuProps>(({ className, mode, theme }) => {
    function generateMenu(menuItems: any) {
        return menuItems
            .filter((menu: MenuAPI) => menu.status === 'Active')
            .map((menu: MenuAPI) => (menu.children ? renderSubMenu(menu) : renderMenuItem(menu)));
    }

    function renderMenuItem(menu: MenuAPI) {
        return (
            <AntMenu.Item key={menu.key}>
                <Link to={menu.url}>
                    {Icon[menu.icon]}
                    <span>{menu.name}</span>
                </Link>
            </AntMenu.Item>
        );
    }

    function renderSubMenu(menu: MenuAPI) {
        let subMenuTitle = (
            <>
                {Icon[menu.icon]}
                <span>{menu.name}</span>
            </>
        );

        return (
            <AntMenu.SubMenu key={menu.key} title={subMenuTitle}>
                {generateMenu(menu.children)}
            </AntMenu.SubMenu>
        );
    }

    return (
        <AntMenu className={className} mode={mode} theme={theme}>
            {generateMenu(menuData)}
        </AntMenu>
    );
});
