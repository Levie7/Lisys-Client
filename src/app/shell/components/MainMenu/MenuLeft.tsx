import { Layout } from 'antd';
import * as React from 'react';
import { Scrollbars } from 'react-custom-scrollbars';

import { Picture } from 'src/shared/components/Picture';

import { logo } from './constants';
import { Menu } from './Menu';

interface MenuLeftProps {
    isMobile: boolean;
}

require('./MenuLeft.sass');

export const MenuLeft = React.memo<MenuLeftProps>(({ isMobile }) => {
    const [onCollapsed, setCollapsed] = React.useState(false);

    function toggleCollapse(type: any) {
        if (type === 'responsive' && onCollapsed) {
            return;
        }

        setCollapsed(!onCollapsed);
    }

    let paramsMobile = {
        collapsed: false,
        collapsible: false,
        onCollapse: toggleCollapse,
        width: 256,
    };
    let paramsDesktop = {
        breakpoint: 'lg',
        collapsed: onCollapsed,
        collapsible: true,
        onCollapse: toggleCollapse,
        width: 256,
    };
    let params = isMobile ? paramsMobile : paramsDesktop;

    return (
        <Layout.Sider {...params} className='MenuLeft'>
            {!onCollapsed && (
                <div className='MenuLeft_Logo d-flex fa-center fj-center'>
                    <Picture
                        alt={logo.light.alt}
                        className='bg-primary'
                        height='46px'
                        src={logo.light.url}
                        title={logo.light.title}
                    />
                </div>
            )}
            <Scrollbars
                autoHide
                style={{ height: isMobile ? 'calc(100vh - 64px)' : 'calc(100vh - 112px)' }}
            >
                <Menu className='MenuLeft_Navigation' mode='inline' theme='dark' />
            </Scrollbars>
        </Layout.Sider>
    );
});
