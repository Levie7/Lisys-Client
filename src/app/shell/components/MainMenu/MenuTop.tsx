import * as React from 'react';

import { Picture } from 'src/shared/components/Picture';

import { logo } from './constants';
import { Menu } from './Menu';

export class MenuTop extends React.Component<{}> {
    render() {
        return (
            <div className='d-flex'>
                <Picture
                    alt={logo.light.alt}
                    className='bg-primary px-4'
                    height='46px'
                    src={logo.light.url}
                    title={logo.light.title}
                />
                <Menu className='w-100' mode='horizontal' theme='dark' />
            </div>
        );
    }
}
