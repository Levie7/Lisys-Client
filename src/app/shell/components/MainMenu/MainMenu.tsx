import * as React from 'react';
import Drawer from 'rc-drawer';

import { UIContext } from 'src/shared/contexts/UIContext';

import { MenuLeft } from './MenuLeft';
import { MenuTop } from './MenuTop';

interface MainMenuState {
    isMenuTop: boolean;
}

require('./MainMenu.sass');
export class MainMenu extends React.Component<{}, MainMenuState> {
    constructor(props: {}) {
        super(props);
        this.state = { isMenuTop: true };
    }

    render() {
        let { isMenuTop } = this.state;

        return (
            <UIContext.Consumer>
                {({ isMobile }) =>
                    isMobile ? (
                        <Drawer className='MainMenu_DarkDrawer' getContainer={null} level={null}>
                            <MenuLeft isMobile={isMobile} />
                        </Drawer>
                    ) : isMenuTop ? (
                        <MenuTop />
                    ) : (
                        <MenuLeft isMobile={isMobile} />
                    )
                }
            </UIContext.Consumer>
        );
    }
}
