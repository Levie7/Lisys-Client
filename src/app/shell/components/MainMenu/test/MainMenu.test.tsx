import { shallow } from 'enzyme';
import React from 'react';

import UIContext, * as UIContextModule from 'src/shared/contexts/UIContext';

import { MainMenu } from '../MainMenu';

describe('MainMenu', () => {
    let context = { isMobile: false };
    let props = {
        isMenuTop: false,
    };
    jest.spyOn(UIContextModule, 'useUIContext').mockImplementation(() => context);
    let wrap = shallow(
        <UIContext.Provider value={context}>
            <MainMenu {...props} />
        </UIContext.Provider>,
        { context }
    );

    it('<MainMenu/> - should render MainMenu component correctly', () => {
        expect(wrap).toMatchSnapshot();
    });

    it('should render main menu component', () => {
        expect(wrap.find('Memo(MainMenuPure)').exists()).toBeTruthy();
    });

    describe('when is mobile site', () => {
        beforeEach(() => {
            context = { isMobile: true };
            wrap.dive().setContext(context);
        });

        it('should render menu left', () => {
            expect(
                wrap
                    .dive()
                    .find('MenuLeft')
                    .exists()
            ).toBeTruthy();
        });
    });

    describe('when is not mobile site', () => {
        beforeEach(() => {
            context = { isMobile: false };
            wrap.dive().setContext(context);
        });

        describe('and is menu top', () => {
            beforeEach(() => {
                props = { isMenuTop: true };
                wrap = shallow(
                    <UIContext.Provider value={context}>
                        <MainMenu {...props} />
                    </UIContext.Provider>,
                    { context }
                ).dive();
            });

            it('should render menu top', () => {
                expect(wrap.find('Memo(MenuTopPure)').exists()).toBeTruthy();
            });
        });

        describe('and is not menu top', () => {
            beforeEach(() => {
                props = { isMenuTop: false };
                wrap = shallow(
                    <UIContext.Provider value={context}>
                        <MainMenu {...props} />
                    </UIContext.Provider>,
                    { context }
                ).dive();
            });

            it('should render menu left', () => {
                expect(wrap.find('MenuLeft').exists()).toBeTruthy();
            });
        });
    });
});
