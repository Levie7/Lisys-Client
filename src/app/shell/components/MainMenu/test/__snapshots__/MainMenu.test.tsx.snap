// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu <MainMenu/> - should render MainMenu component correctly 1`] = `ShallowWrapper {}`;
